capten/config-worker/pkg/activities/provider_git_configure.go

```diff
@@ -0,0 +1,213 @@
+package activities
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/kube-tarian/kad/capten/model"
+	cp "github.com/otiai10/copy"
+	"gopkg.in/yaml.v2"
+)
+
+func (hg *HandleGit) configureCrossplaneProvider(ctx context.Context, params *model.UseCase, pathInRepo, token string) error {
+	gitConfigPlugin := getCICDPlugin()
+
+	// create a dummy directory for creating all the files
+	tempDir, err := os.MkdirTemp(".", "temp*")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDir)
+
+	reqRepo, err := os.MkdirTemp(hg.config.GitCLoneDir, "clone*")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(reqRepo) // clean up
+
+	if err := gitConfigPlugin.Clone(reqRepo, params.RepoURL, token); err != nil {
+		return err
+	}
+
+	if err := createFiles(
+		filepath.Join(tempDir, pathInRepo),
+		params,
+		hg.pluginConfig.GetPluginMap(CrossPlaneProvider)); err != nil {
+		return err
+	}
+
+	// copy the contents to the cloned repo and if dir exists, then replace it
+	err = cp.Copy(
+		filepath.Join(tempDir, pathInRepo),
+		filepath.Join(reqRepo, pathInRepo),
+		cp.Options{
+			OnDirExists: func(src, dest string) cp.DirExistsAction {
+				return cp.Replace
+			}})
+	if err != nil {
+		return err
+	}
+
+	if err := gitConfigPlugin.Commit(".", "configure CrossplaneProvider for the repo",
+		hg.config.GitDefaultCommiterName, hg.config.GitDefaultCommiterEmail); err != nil {
+		return err
+	}
+
+	localBranchName := branchName + "-" + params.Type
+	defaultBranch, err := gitConfigPlugin.GetDefaultBranchName()
+	if err != nil {
+		return err
+	}
+
+	if params.PushToDefaultBranch {
+		localBranchName = defaultBranch
+	}
+
+	if err := gitConfigPlugin.Push(localBranchName, token); err != nil || params.PushToDefaultBranch {
+		return err
+	}
+
+	_, err = createPR(ctx, params.RepoURL, branchName+"-"+params.Type, defaultBranch, token)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createFiles(dir string, params *model.UseCase, pluginMap map[string]string) error {
+
+	// create controllerConfigs
+	for _, providerInfo := range params.CrossplaneProviders {
```

Comment:
create one provider yaml file.

<providername>-provider.yaml contains Provider and ControlConfig resources.

Write separate method to prepare Provider and ControlConfig resources.
Write the resources to file - prepare filename and write